### PR TITLE
feat: prompt for git

### PR DESCRIPTION
**Pull request by LloydTao**
# Feat - Prompt for Git

Issue #587 - Creating a project within an existing repository causes a fatal git error

## Description

This pull request will prompt the user to choose a version control system.

image

There's two main reasons for this:

1. Using `npx create-nuxt-app` in an existing repository fails, as two `.git` directories exist. This is especially prevalent in our projects where we have a Django backend and Nuxt frontend in the same repo.
2. Several types of VCS exist (e.g. SVN), and users shouldn't be forced to delete unwanted directories in order to implement their preference. 

This feature has been designed to easily allow another VCS to be added to the list.

## Changes

The `prompt.js` asks for a VCS preference in the form of a list.

image

The `saofile.js` will now only run `gitInit()` if the respective choice is selected.

image

### CODE REVIEW

**Comment by LloydTao** [comment]
Tests pass on my machine 😣 

image

**Comment by LloydTao** [comment]
Re-linting directly before pushing did the trick. Life of a Windows dev 😁

**Comment by LloydTao** [comment]
I'm now realising that these tests are auto-generated 😂

**Comment by LloydTao** [comment]
All good to review now 😜

File: packages/create-nuxt-app/lib/prompts.js [hunk 1]
```diff
@@ -112,5 +112,15 @@ module.exports = [
...
+  },
+  {
+    name: 'vcs',
+    message: 'Version control system:',
+    type: 'list',
+    choices: [
+      { name: 'None', value: 'none' },
+      { name: 'Git', value: 'git' }
+    ],
+    default: 'none'
```
Comment (pi0) [inline packages/create-nuxt-app/lib/prompts.js:124]: Maybe it would be better to keep default as `git` like before? :)

Comment (LloydTao) [reply packages/create-nuxt-app/lib/prompts.js]: @pi0 sure! my reasoning here was that the other fields use `default: 'none'`, so it was an attempt to match those design patterns, but i understand why Git is a good exception 😼

the prompt now uses `git` as default, as well as putting that option at the top of the list

- Use Git as default VCS: https://github.com/nuxt/create-nuxt-app/pull/596/commits/2924cc836eaa0ee59c61a5658797c349e54ccad2

**Review by pi0 — APPROVED** [review]
LGTM. Thanks :heart:

**Comment by LloydTao** [comment]
thank you. this was my first open source contribution to a mainstream repo, so cheers for the good experience 😼

**Pull request merged**